146 - Brian Diep - Accounts Receivable >> Sales Orders

Bdiep6 opened the pull request:
## Blazor Pages Created:

- SalesOrders.razor - Display list of all sales orders with viewing and management options
- NewSalesOrder.razor - Form to create new sales orders with customer selection and order details
- SalesOrder.razor - Detail/edit view for individual sales orders

## View Wrappers Updated:

- AddSalesOrder.cshtml - Injects NewSalesOrder Blazor component
- SalesOrder.cshtml - Injects SalesOrder Blazor component
- salesorders.cshtml - Injects SalesOrders Blazor component

Bdiep6 committed:
Add BlazorGDB to AppHost in Program.cs and src.AppHost.csproj

Bdiep6 committed:
Fix sales orders functionality

- Fix SalesOrders.razor to properly display total amounts in the list view
- Add NewSalesOrder.razor component for creating new sales orders
- Implement proper API integration with customer/item dropdowns
- Clean up unnecessary backend changes, keeping only essential frontend fixes

Bdiep6 committed:
Remove automatic navigation after saving sales order

- Save Order button now saves and stays on page instead of auto-redirecting
- Users can review saved order and manually navigate back using Back to List button

Bdiep6 committed:
Migrate Sales Orders Page and Add New Sales Order Page to Blazor components.

Bdiep6 committed:
Add SalesOrder view component and improve SalesOrders row selection

Bdiep6 committed:
Added SalesOrder Blazor component and update routing

- Created SalesOrder.razor component
- Updated SalesOrder.cshtml to use Blazor component instead of MVC view

NOTE: SalesOrder endpoint returns 500 error due to nullable issue in API
  (CustomerName lookup fails when customer data is unavailable)
  Fallback logic can be added if needed, or API can be fixed to handle null customers.
  Checked on the hosted site, https://gdbmvc.azurewebsites.net/SalesOrder?id=1 also returns 404.

Bdiep6 committed:
Add note to SalesOrder debug panel about nullable field errors

medhatelmasry commented:
1) When an item is highlighted in the list, it would be good to change the color of the row so that the user understands that the row has been selected.

2) After selecting an item >> click on View: I get this error:
<img width="1098" height="420" alt="image" src="https://github.com/user-attachments/assets/c50deb70-966f-4487-8d56-7064248a5bd7" />

3) After selecting an item >> click on "Create Invoice": I get this error:
<img width="1229" height="394" alt="image" src="https://github.com/user-attachments/assets/2f83048f-c481-4147-b947-f4229db9b29a" />

Bdiep6 committed:
Added UI to selected row in Sales Orders (list),fixed page for Sales Order (singular), Added Blazor component for SalesInvoice and saves to list

Bdiep6 commented:
## Sales Order Minor UI Improvements

- Added background color and bold font to selected row for better visual feedback
- SaveOrder and Cancel buttons now navigate back to the SalesOrders page so users can verify changes
- Removed redundant "Back to list" button

## Bug Fixes

"After selecting an item >> click on View: I get this error"
- Issue in SalesController.cs: Forced unwrapping of nullable fields (Status, CustomerId) without null checks which caused it to access properties on null response without validation. Fixed by replacing strict validation with null coalescing (??) operators

"After selecting an item >> click on "Create Invoice": I get this error"
-  Issue in previous MVC impementation. Added a new Blazor component for creating and managing sales invoices, replacing the MVC form

medhatelmasry commented:
Works well. PR was accepted and branch merged.

Regards,
Medhat Elmasry



On Wed, Nov 26, 2025 at 9:00 PM Bdiep6 ***@***.***> wrote:

> *Bdiep6* left a comment (medhatelmasry/GoodBooks#178)
> <https://github.com/medhatelmasry/GoodBooks/pull/178#issuecomment-3584211049>
> Sales Order Minor UI Improvements
>
>    - Added background color and bold font to selected row for better
>    visual feedback
>    - SaveOrder and Cancel buttons now navigate back to the SalesOrders
>    page so users can verify changes
>    - Removed redundant "Back to list" button
>
> Bug Fixes
>
> "After selecting an item >> click on View: I get this error"
>
>    - Issue in SalesController.cs: Forced unwrapping of nullable fields
>    (Status, CustomerId) without null checks which caused it to access
>    properties on null response without validation. Fixed by replacing strict
>    validation with null coalescing (??) operators
>
> "After selecting an item >> click on "Create Invoice": I get this error"
>
>    - Issue in previous MVC impementation. Added a new Blazor component
>    for creating and managing sales invoices, replacing the MVC form
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/medhatelmasry/GoodBooks/pull/178#issuecomment-3584211049>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN4UQS7MAHGCSKUPVYZXYD362APTAVCNFSM6AAAAACNHITVGSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTKOBUGIYTCMBUHE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
